tests/Unit/Helpers/Evolution/DgSubcell/ProjectionTestHelpers.cpp

```diff
@@ -0,0 +1,157 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Helpers/Evolution/DgSubcell/ProjectionTestHelpers.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace TestHelpers::evolution::dg::subcell {
+// computes a simple polynomial over the grid that we then project and
+// reconstruct in the tests.
+template <size_t Dim>
+DataVector cell_values(
+    const size_t max_polynomial_degree_plus_one,
+    const tnsr::I<DataVector, Dim, Frame::Logical>& coords) noexcept {
+  DataVector result(get<0>(coords).size(), 0.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    for (size_t i = 0; i < max_polynomial_degree_plus_one; ++i) {
+      result += pow(coords.get(d), i);
+    }
+  }
+  return result;
+}
+
+// Computes the average in each finite volume cell multiplied by the cell's
+// volume.
+template <size_t Dim>
+DataVector cell_averages_times_volume(
+    const Mesh<Dim>& mesh, const Index<Dim>& subcell_extents) noexcept {
```

Comment:
I think this Mesh could also be a `size_t max_polynomial_degree_plus_one`

Reply:
Okay, pushed...